src/krb5/_creds.pyx

```diff
@@ -97,6 +167,68 @@ cdef class Creds:
     def __str__(Creds self) -> str:
         return "Creds"
 
+    @property
+    def client(Creds self) -> Principal:
+        princ = Principal(self.ctx, 0, needs_free=0)
```

Comment:
We should probably make a copy of the principal here (same with server). If we don't make a copy then trying to access any of the principal attributes after this credential has been deallocated will most likely cause a seg fault.

Reply:
Ok, I'll do that.